src/pint/residuals.py

```diff
@@ -370,11 +370,17 @@ def calc_phase_resids(
             # and delta_pulse_numbers (from PHASE lines or adding phase jumps in GUI)
             i = pulse_num.copy()
             f = np.zeros_like(pulse_num)
-            if np.any(np.isnan(pulse_num)):
+
+            c = np.isnan(pulse_num)
+            if np.any(c):
```

Comment:
What is the purpose of this?  Won't the first exception mean you never get to the later code?

Reply:
I presume you are referring to the `if pulse_num is None` clause.
That clause checks if the pulse number column is missing altogether from the TOAs object, and raises an exception immediately if it is not found.

The `if np.any(c)` clause is checking if any of the pulse numbers are `nan` when the column itself is present.

Reply:
right.  but then you have another `if np.any(c)` down below (L381) but it would raise an exception here.

Reply:
Ah! that's not required. I have removed it.